apps/prairielearn/src/middlewares/clientFingerprint.ts

```diff
@@ -0,0 +1,46 @@
+import asyncHandler = require('express-async-handler');
+import * as sqldb from '@prairielearn/postgres';
+import { IdSchema } from '../lib/db-types';
+
+const sql = sqldb.loadSqlEquiv(__filename);
+
+export default asyncHandler(async (req, res, next) => {
+  const user_session_id = await sqldb.queryOptionalRow(
+    sql.select_session_id,
+    { user_session_id: req.session.id },
```

Comment:
Our `user_sessions` table is horribly confusing, sorry in advance about this. What we're actually selecting here is `user_session.id`, so I'd call this query `select_user_session_id`. The value we're passing in here corresponds to the `session_id` column, so I'd call the parameter `session_id`. Does that make sense?